<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo">
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Peer Assignment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a SQLIte database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. Click on the links below to download and save the datasets (.CSV files):

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Census Data</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Public Schools</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Crime Data</a>

**NOTE:** Ensure you have downloaded the datasets using the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


### Store the datasets in database tables

To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables in as under:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**

Let us now load the ipython-sql  extension and establish a connection with the database

* Here you will be loading the csv files into the pandas Dataframe and then loading the data into the above mentioned sqlite tables.

* Next you will be connecting to the sqlite database  **FinalDB**.

Refer to the previous lab for hints .

<a href ="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/Module%205/DB0201EN-Week3-1-4-Analyzing_SQLite.ipynb">Hands-on Lab: Analyzing a real World Data Set</a>





In [2]:
%load_ext sql

In [3]:
import csv
import pandas as pd
import sqlite3


con = sqlite3.connect("finalDB.db")
cur = con.cursor()

In [4]:
!pip install -q pandas==1.1.5

In [5]:
# connect to sqlite
%sql sqlite:///finalDB.db

'Connected: @finalDB.db'

In [25]:
# Load data from CSV file into a Pandas dataframe
df_census = pd.read_csv('https://data.cityofchicago.org/resource/kn9c-c2s2.csv')
df_crime = pd.read_csv("C:\PERSONAL\02.1. MBAN Preparation\COURSERA\IBM Data Science_Data Analyst Certificate\4. Databases and SQL for Data Science with Python\ChicagoCrimeData.csv")
df_school = pd.read_csv('https://data.cityofchicago.org/resource/9xs2-f89t.csv')


# Create a SQLite database connection
conn = sqlite3.connect('finalDB.db')

# Create a table in the database using the Pandas dataframe
df_census.to_sql('chicago_census_data', conn, if_exists='replace', index=False)
df_crime.to_sql('chicago_crime_data', conn, if_exists='replace', index=False)
df_school.to_sql('chicago_school_data', conn, if_exists='replace', index=False)

# Close the database connection
conn.close()


FileNotFoundError: [Errno 2] No such file or directory: 'C:\\PERSONAL\x02.1. MBAN Preparation\\COURSERA\\IBM Data Science_Data Analyst Certificate\x04. Databases and SQL for Data Science with Python\\ChicagoCrimeData.csv'

In [26]:
%sql SELECT * FROM chicago_census_data limit 5;


 * sqlite:///finalDB.db
Done.


[(1.0, 'Rogers Park', 7.7, 23.6, 8.7, 18.2, 27.5, 23939, 39.0),
 (2.0, 'West Ridge', 7.8, 17.2, 8.8, 20.8, 38.5, 23040, 46.0),
 (3.0, 'Uptown', 3.8, 24.0, 8.9, 11.8, 22.2, 35787, 20.0),
 (4.0, 'Lincoln Square', 3.4, 10.9, 8.2, 13.4, 25.5, 37524, 17.0),
 (5.0, 'North Center', 0.3, 7.5, 5.2, 4.5, 26.2, 57123, 6.0)]

In [27]:
%sql SELECT * FROM chicago_crime_data limit 5;

 * sqlite:///finalDB.db
Done.


[(13046690, 'JG228930', '2023-04-18T23:59:00.000', '026XX S KILDARE AVE', '143A', 'WEAPONS VIOLATION', 'UNLAWFUL POSSESSION - HANDGUN', 'ALLEY', 0, 0, 1031, 10, 22, 30, '15', 1148147.0, 1886130.0, 2023, '2023-04-25T16:47:28.000', 41.843484949, -87.7318287, '\n,  \n(41.843484949, -87.7318287)'),
 (13046700, 'JG228905', '2023-04-18T23:47:00.000', '0000X W ROOSEVELT RD', '143A', 'WEAPONS VIOLATION', 'UNLAWFUL POSSESSION - HANDGUN', 'STREET', 1, 0, 123, 1, 4, 32, '15', 1176326.0, 1895061.0, 2023, '2023-04-25T16:47:28.000', 41.867404119, -87.628149145, '\n,  \n(41.867404119, -87.628149145)'),
 (13046696, 'JG228904', '2023-04-18T23:32:00.000', '025XX N SPRINGFIELD AVE', '0486', 'BATTERY', 'DOMESTIC BATTERY SIMPLE', 'APARTMENT', 1, 1, 2524, 25, 31, 22, '08B', 1149955.0, 1916596.0, 2023, '2023-04-25T16:47:28.000', 41.927052012, -87.724400318, '\n,  \n(41.927052012, -87.724400318)'),
 (13046900, 'JG229136', '2023-04-18T23:30:00.000', '064XX S ELLIS AVE', '0910', 'MOTOR VEHICLE THEFT', 'AUTOMOBILE', 'STREET', 0, 0, 314, 3, 20, 42, '07', 1184161.0, 1862588.0, 2023, '2023-04-25T16:47:28.000', 41.778115781, -87.600403043, '\n,  \n(41.778115781, -87.600403043)'),
 (13047932, 'JG230321', '2023-04-18T23:30:00.000', '048XX S LOOMIS BLVD', '0810', 'THEFT', 'OVER $500', 'APARTMENT', 0, 1, 933, 9, 20, 61, '06', 1167817.0, 1872590.0, 2023, '2023-04-25T16:47:28.000', 41.805929086, -87.660033433, '\n,  \n(41.805929086, -87.660033433)')]

In [ ]:
%sql SELECT * FROM chicago_school_data limit 5;

 * sqlite:///finalDB.db
Done.


[(610038, 'Abraham Lincoln Elementary School', 'ES', '615 W Kemper Pl ', 'Chicago', 'IL', 60614, '(773) 534-5720', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610038.pdf', 'Fullerton Elementary Network', 'NORTH-NORTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 1', 'Yes', 'Very Strong', 99.0, 'Very Strong', '99', 'Strong', 74.0, 'Strong', 66.0, 'Strong', '65', 'Strong', '70', 'Strong', '56', 'Average', '47', 96.0, 2.0, 96.4, 95.8, '80.1', '43.3', '89.6', '84.9', '60.7', '62.6', '81.9', '85.2', '52', '62.4', '66.3', '77.9', 69.7, 64.4, 0.2, 0.9, 'Yellow', 'Green', '67.1', '54.5', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 813, 33, 'NDA', 150000000000000, 1171699.458, 1915829.428, 41.92449696, -87.64452163, 7, 'LINCOLN PARK', 43, 18, '(41.92449696, -87.64452163)'),
 (610281, 'Adam Clayton Powell Paideia Community Academy Elementary School', 'ES', '7511 S South Shore Dr ', 'Chicago', 'IL', 60649, '(773) 535-6650', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610281.pdf', 'Skyway Elementary Network', 'SOUTH SIDE COLLABORATIVE', 'No', 'Track_E', 'Not on Probation', 'Level 1', 'No', 'Average', 54.0, 'Strong', '66', 'Strong', 74.0, 'Very Strong', 84.0, 'Strong', '63', 'Strong', '76', 'Weak', '46', 'Average', '50', 95.6, 15.7, 95.3, 100.0, '62.4', '51.7', '21.9', '15.1', '29', '42.8', '38.5', '27.4', '44.8', '42.7', '14.1', '34.4', 16.8, 16.5, 0.7, 1.4, 'Green', 'Green', '17.2', '27.3', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 521, 46, 'NDA', 150000000000000, 1196129.985, 1856209.466, 41.76032435, -87.55673627, 43, 'SOUTH SHORE', 7, 4, '(41.76032435, -87.55673627)'),
 (610185, 'Adlai E Stevenson Elementary School', 'ES', '8010 S Kostner Ave ', 'Chicago', 'IL', 60652, '(773) 535-2280', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610185.pdf', 'Midway Elementary Network', 'SOUTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 2', 'No', 'Strong', 61.0, 'NDA', 'NDA', 'Average', 50.0, 'Weak', 36.0, 'NDA', 'NDA', 'NDA', 'NDA', 'Average', '47', 'Weak', '41', 95.7, 2.3, 94.7, 98.3, '53.7', '26.6', '38.3', '34.7', '43.7', '57.3', '48.8', '39.2', '46.8', '44', '7.5', '21.9', 18.3, 15.5, -0.9, -1.0, 'Red', 'Red', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 1324, 44, 'NDA', 150000000000000, 1148427.165, 1851012.215, 41.74711093, -87.73170248, 70, 'ASHBURN', 13, 8, '(41.74711093, -87.73170248)'),
 (609993, 'Agustin Lara Elementary Academy', 'ES', '4619 S Wolcott Ave ', 'Chicago', 'IL', 60609, '(773) 535-4389', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_609993.pdf', 'Pershing Elementary Network', 'SOUTHWEST SIDE COLLABORATIVE', 'No', 'Track_E', 'Not on Probation', 'Level 1', 'No', 'Average', 56.0, 'Average', '44', 'Average', 45.0, 'Weak', 37.0, 'Strong', '65', 'Average', '48', 'Average', '53', 'Strong', '58', 95.5, 10.4, 95.8, 100.0, '76.9', 'NDA', '26', '24.7', '61.8', '49.7', '39.2', '27.2', '69.7', '60.6', '9.1', '18.2', 11.1, 9.6, 0.9, 2.4, 'Green', 'Green', '42.9', '25', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 556, 42, 'NDA', 150000000000000, 1164504.29, 1873959.199, 41.8097569, -87.6721446, 61, 'NEW CITY', 20, 9, '(41.8097569, -87.6721446)'),
 (610513, 'Air Force Academy High School', 'HS', '3630 S Wells St ', 'Chicago', 'IL', 60609, '(773) 535-1590', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610513.pdf', 'Southwest Side High School Network', 'SOUTHWEST SIDE COLLABORATIVE', 'NDA', 'Standard', 'Not on Probation', 'Not Enough Data', 'Yes', 'Average', 49.0, 'Strong', '60', 'Strong', 60.0, 'Average', 55.0, 'Average', '45', 'Average', '54', 'Average', '53', 'Average', '49', 93.3, 15.6, 96.9, 100.0, 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', None, None, None, None, 'NDA', 'NDA', 'NDA', 'NDA', '14.6', '14.8', 'NDA',

## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [29]:
%sql SELECT COUNT(*) FROM chicago_crime_data;

 * sqlite:///finalDB.db
Done.


[(1000,)]

### Problem 2

##### List community areas with per capita income less than 11000.


In [30]:
%sql SELECT community_area_name FROM chicago_census_data WHERE per_capita_income_ < 11000;

 * sqlite:///finalDB.db
Done.


[('West Garfield Park',),
 ('South Lawndale',),
 ('Fuller Park',),
 ('Riverdale',)]

### Problem 3

##### List all case numbers for crimes  involving minors?(children are not considered minors for the purposes of crime analysis)


In [31]:
%sql SELECT case_number FROM chicago_crime_data WHERE description LIKE "%MINOR";

 * sqlite:///finalDB.db
Done.


[('JG228737',)]

In [32]:
%sql SELECT DISTINCT primary_type FROM chicago_crime_data;

 * sqlite:///finalDB.db
Done.


[('WEAPONS VIOLATION',),
 ('BATTERY',),
 ('MOTOR VEHICLE THEFT',),
 ('THEFT',),
 ('BURGLARY',),
 ('CRIMINAL DAMAGE',),
 ('OTHER OFFENSE',),
 ('ASSAULT',),
 ('PROSTITUTION',),
 ('ROBBERY',),
 ('DECEPTIVE PRACTICE',),
 ('INTERFERENCE WITH PUBLIC OFFICER',),
 ('ARSON',),
 ('CRIMINAL TRESPASS',),
 ('CRIMINAL SEXUAL ASSAULT',),
 ('OFFENSE INVOLVING CHILDREN',),
 ('SEX OFFENSE',),
 ('LIQUOR LAW VIOLATION',),
 ('NARCOTICS',),
 ('HOMICIDE',),
 ('PUBLIC PEACE VIOLATION',)]

### Problem 4

##### List all kidnapping crimes involving a child?


In [36]:
%sql SELECT case_number,primary_type, description  FROM chicago_crime_data WHERE primary_type = "OFFENSE INVOLVING CHILDREN";

 * sqlite:///finalDB.db
Done.


[('JG231849', 'OFFENSE INVOLVING CHILDREN', 'CHILD ABUSE'),
 ('JG228570', 'OFFENSE INVOLVING CHILDREN', 'OTHER OFFENSE'),
 ('JG229431', 'OFFENSE INVOLVING CHILDREN', 'CHILD ABUSE'),
 ('JG229508', 'OFFENSE INVOLVING CHILDREN', 'ENDANGERING LIFE / HEALTH OF CHILD'),
 ('JG228343', 'OFFENSE INVOLVING CHILDREN', 'CHILD ABUSE'),
 ('JG228858', 'OFFENSE INVOLVING CHILDREN', 'CHILD ABUSE'),
 ('JG227974', 'OFFENSE INVOLVING CHILDREN', 'CONTRIBUTE TO THE CRIMINAL DELINQUENCY OF CHILD'),
 ('JG227473', 'OFFENSE INVOLVING CHILDREN', 'CRIMINAL SEXUAL ABUSE BY FAMILY MEMBER'),
 ('JG227043', 'OFFENSE INVOLVING CHILDREN', 'CHILD ABUSE'),
 ('JG232077', 'OFFENSE INVOLVING CHILDREN', 'AGGRAVATED SEXUAL ASSAULT OF CHILD BY FAMILY MEMBER')]

### Problem 5

##### What kinds of crimes were recorded at schools?


In [37]:
%sql SELECT DISTINCT primary_type FROM chicago_crime_data WHERE location_description LIKE "%SCHOOL%";

 * sqlite:///finalDB.db
Done.


[('CRIMINAL TRESPASS',),
 ('ASSAULT',),
 ('OTHER OFFENSE',),
 ('BATTERY',),
 ('PUBLIC PEACE VIOLATION',),
 ('THEFT',),
 ('SEX OFFENSE',)]

### Problem 6

##### List the average safety score for each type of school.


In [38]:
%sql SELECT DISTINCT elementary_or_high_school, ROUND(AVG(safety_score),2) \
FROM chicago_school_data \
GROUP BY elementary_or_high_school;

 * sqlite:///finalDB.db
Done.


[('ES', 49.52), ('HS', 49.62), ('MS', 48.0)]

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [39]:
%sql SELECT community_area_name, percent_households_below_poverty  FROM chicago_census_data ORDER BY percent_households_below_poverty DESC LIMIT 5;

 * sqlite:///finalDB.db
Done.


[('Riverdale', 56.5),
 ('Fuller Park', 51.2),
 ('Englewood', 46.6),
 ('North Lawndale', 43.1),
 ('East Garfield Park', 42.4)]

### Problem 8

##### Which community area is most crime prone?


In [43]:
%sql SELECT community_area, COUNT(community_area) AS number_of_crimes FROM chicago_crime_data GROUP BY community_area ORDER BY number_of_crimes DESC LIMIT 1;

 * sqlite:///finalDB.db
Done.


[(25, 48)]

Double-click **here** for a hint

<!--
Query for the 'community area number' that is most crime prone.
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [19]:
%sql SELECT community_area_name FROM chicago_census_data WHERE hardship_index = (SELECT MAX(hardship_index) FROM chicago_census_data);


 * sqlite:///finalDB.db
Done.


[('Riverdale',)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [47]:
%sql SELECT community_area_name FROM chicago_census_data WHERE community_area = (SELECT community_area FROM chicago_crime_data GROUP BY community_area ORDER BY COUNT(community_area) DESC) LIMIT 1);


 * sqlite:///finalDB.db
(sqlite3.OperationalError) near ")": syntax error
[SQL: SELECT community_area_name FROM chicago_census_data WHERE community_area = (SELECT community_area FROM chicago_crime_data GROUP BY community_area ORDER BY COUNT(community_area) DESC) LIMIT 1);]
(Background on this error at: http://sqlalche.me/e/13/e3q8)


Copyright © 2020 This notebook and its source code are released under the terms of the [MIT License](https://bigdatauniversity.com/mit-license?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ).


## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>

## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
| 2022-03-04 | 2.5     | Lakshmi Holla     | Changed markdown.                   |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>
